### DISCUSSION TITLE
Set custom cookie response

### DISCUSSION THREAD

**Discussion by xavierbeillas** [graphql/graphql-http#105] [Q&A] [answered]
I want to set a new cookie in a resolver response.

I was expecting to pass down server response from a custom context but only getting `req` et `params`. No `res`.
How would you achieve custom response cookie ?

Thanks.

**Comment by enisdenjo** [comment, marked as answer]
`graphql-http` core handler is pure, it takes a request and returns a response. This response can be manipulated to your desires.

But, I guess you're using one of the `use/` handlers. So, here's a few examples:

- When using [`use/http`](https://github.com/graphql/graphql-http#with-http) the [ServerResponse is available in the request context](https://github.com/graphql/graphql-http/blob/823bf115785df5382284573ff830755881090a6c/src/use/http.ts#L8-L15) (`req.context.res`)
  - Same for [`use/http2`](https://github.com/graphql/graphql-http#with-http2)
  - Same for [`use/express`](https://github.com/graphql/graphql-http#with-express)
  - Same for [`use/fastify`](https://github.com/graphql/graphql-http#with-fastify), but as a `FastifyReply`
  - Same for [`use/koa`](https://github.com/graphql/graphql-http#with-koa)
- When using [`use/deno`](https://github.com/graphql/graphql-http#with-deno), the handler is also pure, so you'd just add headers to the returned response.
  - Same for [`use/bun`](https://github.com/graphql/graphql-http#with-bun)
  - Same for [`use/@netlify/functions`](https://github.com/graphql/graphql-http#with-netlify-functions)

**Reply by xavierbeillas** [reply]
I didn't see `res` was already included in `req.context`. I was limited to look for something like `context: async (req, res, args) => {`.
Thanks a lot (and specially for your reactivity :) )
